mdbackup/storage/s3.py

```diff
@@ -72,27 +71,25 @@ def upload(self, path: Path, parent: Union[Path, str, str] = None):
             key = path.name
         key = self.__ok_key(key)
         self.__log.info(f'Uploading file {key} (from {path})')
-        ret = self.__s3.upload_file(str(path.absolute()),
-                                    self.__bucket,
-                                    key,
-                                    ExtraArgs={
-                                        'ContentType': magic.from_file(str(path.absolute()), mime=True),
-                                        'StorageClass': self.__storageclass,
-                                    })
-        self.__log.debug(ret)
+        self.__s3.upload_file(str(path.absolute()),
+                              self.__bucket,
+                              key,
+                              ExtraArgs={
+                                  'ContentType': magic.from_file(str(path.absolute()), mime=True),
+                                  'StorageClass': self.__storageclass,
+                              })
 
     def delete(self, path: Union[Path, str]):
-        path = self.__ok_key(path)
-        self.__log.info(f'Deleting {self.__pre}/{path}')
+        full_path = self.__ok_key(path)
+        self.__log.info(f'Deleting {full_path}')
         objects_to_delete = self.list_directory_recursive(path)[::-1]
         while len(objects_to_delete) > 0:
-            ret = self.__s3.delete_objects(
+            self.__s3.delete_objects(
                 Bucket=self.__bucket,
                 Delete={
-                    'Objects': [{'Key': self.__ok_key(key)} for key in objects_to_delete],
+                    'Objects': [{'Key': f'{self.__pre}/{key}'} for key in objects_to_delete],
```

Comment:
The `__ok_key()` method removed the final `/` and folders were not removed. Now they will be removed properly :)